src/main/java/org/jenkinsci/plugins/gitclient/JGitAPIImpl.java

```diff
@@ -163,16 +182,76 @@ public class JGitAPIImpl extends LegacyCompatibleGitAPIImpl {
          * CliGitAPIImpl behavior */
         super(workspace == null ? new File(".") : workspace, hostKeyFactory);
         this.listener = listener;
+        hostKeyVerifierFactory = hostKeyFactory;
+    }
 
-        // to avoid rogue plugins from clobbering what we use, always
-        // make a point of overwriting it with ours.
-        SshSessionFactory.setInstance(new TrileadSessionFactory(hostKeyFactory, listener));
-
-        if (httpConnectionFactory != null) {
-            httpConnectionFactory.setCredentialsProvider(asSmartCredentialsProvider());
-            // allow override of HttpConnectionFactory to avoid JENKINS-37934
-            HttpTransport.setConnectionFactory(httpConnectionFactory);
+    public SshdSessionFactory buildSshdSessionFactory(@NonNull final HostKeyVerifierFactory hostKeyVerifierFactory) {
+        if (Files.notExists(hostKeyVerifierFactory.getKnownHostsFile().toPath())) {
+            try {
+                Files.createDirectories(hostKeyVerifierFactory
+                        .getKnownHostsFile()
+                        .getParentFile()
+                        .toPath());
+                Files.createFile(hostKeyVerifierFactory.getKnownHostsFile().toPath());
```

Comment:
non secured permissions. not sure what is sensitive in a `known_hosts` file.
if there is default umask with write permission, there are probably even more issues than a  `known_hosts` file :)
btw this has been the case for ages now with the trilead API. not sure why we should changed that.

Reply:
nothing in the `known_hosts` is sensitive, however the `.ssh` directory is and if you create this insecurely then other commands may fail later (ie it could break the build, or leave an agent unable to be used later due to this).

Reply:
I know this, but the problem is to change a behavior that has existed for a long time with the current code.
if everybody agrees I don't really mind but it's a change of the result on the files created

Reply:
`man ssh` says 
```
     ~/.ssh/
             This directory is the default location for all user-specific configuration and authentication information.  There is no general requirement to keep the en‐
             tire contents of this directory secret, but the recommended permissions are read/write/execute for the user, and not accessible by others.
```

https://linux.die.net/man/1/ssh

Reply:
> `man ssh` says
> 
> ```
>      ~/.ssh/
>              This directory is the default location for all user-specific configuration and authentication information.  There is no general requirement to keep the en‐
>              tire contents of this directory secret, but the recommended permissions are read/write/execute for the user, and not accessible by others.
> ```
> 
> https://linux.die.net/man/1/ssh

and yet the source code does checking...
https://github.com/openssh/openssh-portable/blob/a8fbe2f7d0d96d299ee8e69769e3b51067978748/regress/check-perm.c#L129-L135

regardless, this is going against the recommendations you point out, and this should not be doing that unless there are valid reasons.
If this was an existing behaviour then it does not need to be changed here - I filed https://issues.jenkins.io/browse/JENKINS-73305 for tracking.

Reply:
`there is no general requirement... but recommendation` :)
Not sure when this check has been added but I have this locally and `git` works fine
```
olamy@pop-os:~$ ls -alrt ~/ | grep .ssh
drwxrwxr-x  2 olamy olamy     4096 Apr 29 16:01 .ssh_backup
drwxrwxr-x  2 olamy olamy     4096 Jun 13 18:30 .ssh
-rw-------  1 olamy olamy      339 Jun 13 18:32 .lesshst
olamy@pop-os:~$ ls -alrt ~/.ssh/
total 324
-rw-------  1 olamy olamy    381 Jun  1  2023 id_rsa.pub
-rw-------  1 olamy olamy   1675 Jun  1  2023 id_rsa
-rw-------  1 olamy olamy    611 Jun  1  2023 id_dsa.pub
-rw-------  1 olamy olamy    609 Jun  1  2023 id_dsa_apache.pub
-rw-------  1 olamy olamy    736 Jun  1  2023 id_dsa_apache
-rw-------  1 olamy olamy    668 Jun  1  2023 id_dsa
-rw-------  1 olamy olamy    387 Jun  1  2023 github_rsa.pub
-rw-------  1 olamy olamy   1766 Jun  1  2023 github_rsa
-rw-------  1 olamy olamy   1226 Jun  1  2023 authorized_keys
-rw-------  1 olamy olamy    566 Apr 29 16:01 id_rsa_git_client.pub
-rw-------  1 olamy olamy   2602 Apr 29 16:01 id_rsa_git_client
-rw-------  1 olamy olamy    319 Jun 10 12:16 config
-rw-rw-r--  1 olamy olamy 274069 Jun 11 13:39 known_hosts
drwxrwxr-x  2 olamy olamy   4096 Jun 13 18:30 .
drwxr-x--- 71 olamy olamy   4096 Jun 13 18:32 ..
olamy@pop-os:~$ ssh -V
OpenSSH_8.9p1 Ubuntu-3ubuntu0.7, OpenSSL 3.0.2 15 Mar 2022
```